### PR TITLE
🧪 📒 Add `nbmake` to test the Jupyter notebooks

### LINKED ISSUE #881: Documentation improvement: Installation and notebooks
### Describe the bug

Installing the source files using the <pip install -e .> I got an error with the opencv-python installation, adding <python -m pip install --upgrade pip wheel setuptools> before could solve the problem.

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

python -m venv anonmalib_env
anonmalib_env\Scripts\activate
git clone https://github.com/openvinotoolkit/anomalib.git
cd anomalib
pip install -e .
image

### OS information

OS: Win 10
Python: 3.8
Anomalib = latest
No-conda environment. Instead <venv>

### Expected behavior

adding <python -m pip install --upgrade pip wheel setuptools>, fixes the issue.

### Screenshots

_No response_

### Pip/GitHub

GitHub

### What version/branch did you use?

_No response_

### Configuration YAML

```yaml
N/A
```

### Logs

```shell
N/A
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description

- Provide a summary of the modification as well as the issue that has been resolved. List any dependencies that this modification necessitates.

- Fixes CVS-101984
- Fixes #881

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

File: notebooks/000_getting_started/001_getting_started.ipynb [hunk 1]
```diff
@@ -23,9 +23,15 @@
    "cell_type": "markdown",
```
[the remark this answers is not in the document]
Comment (samet-akcay) [reply notebooks/000_getting_started/001_getting_started.ipynb:23]: A disadvantage is that we would have the following structure if we are working from the notebooks directory 

anomalib/notebooks/000_getting_started/anomalib/..

**Review by ashwinvaidya17 — COMMENTED** [review]
Maybe we should look into having a helper function in notebooks folder. If we add more notebooks then this might become unmanageable. I am not sure what's the best way to avoid code duplication.

**Comment by samet-akcay** [comment]
> Maybe we should look into having a helper function in notebooks folder. If we add more notebooks then this might become unmanageable. I am not sure what's the best way to avoid code duplication.

Would we still have this problem if the. notebooks only use API? Like no configs, no reference to the repo etc.?

**Comment by ashwinvaidya17** [comment]
I don't think reviewNB highlighted my code currently. Even if we only use the API without reference to the repo we would still need reference to the dataset path.
What if we move this line to
```python
current_directory = Path.cwd()
if current_directory.name == "300_benchmarking":
    # On the assumption that, the notebook is located in
    #   ~/anomalib/notebooks/300_benchmarking/
    root_directory = current_directory.parent.parent
elif current_directory.name == "anomalib":
    # This means that the notebook is run from the main anomalib directory.
    root_directory = current_directory
else:
    # Otherwise, we'll need to clone the anomalib repo to the `current_directory`
    repo = Repo.clone_from(url="https://github.com/openvinotoolkit/anomalib.git", to_path=current_directory)
    root_directory = current_directory / "anomalib"

os.chdir(root_directory)
```
to a method in helpers and do something like
```python
from helpers import get_root_dir

root_directory = get_root_dir(current_dir_name="300_benchmarking")
os.chdir(root_directory)
```

**Comment by samet-akcay** [comment]
> I don't think reviewNB highlighted my code currently. Even if we only use the API without reference to the repo we would still need reference to the dataset path. What if we move this line to
> 
> ```python
> current_directory = Path.cwd()
> if current_directory.name == "300_benchmarking":
>     # On the assumption that, the notebook is located in
>     #   ~/anomalib/notebooks/300_benchmarking/
>     root_directory = current_directory.parent.parent
> elif current_directory.name == "anomalib":
>     # This means that the notebook is run from the main anomalib directory.
>     root_directory = current_directory
> else:
>     # Otherwise, we'll need to clone the anomalib repo to the `current_directory`
>     repo = Repo.clone_from(url="https://github.com/openvinotoolkit/anomalib.git", to_path=current_directory)
>     root_directory = current_directory / "anomalib"
> 
> os.chdir(root_directory)
> ```
> 
> to a method in helpers and do something like
> 
> ```python
> from helpers import get_root_dir
> 
> root_directory = get_root_dir(current_dir_name="300_benchmarking")
> os.chdir(root_directory)
> ```

where would we store `helpers` directory? Can we call it from a Google Colab notebook?

**Comment by ashwinvaidya17** [comment]
Good point. I was thinking of running the notebook locally. In that case we have no option but to keep this.

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks again

**Pull request merged**